qiskit/qasm3/exporter.py

```diff
@@ -1129,6 +1142,15 @@ def build_for_loop(self, instruction: CircuitInstruction) -> ast.ForLoopStatemen
             body_ast = ast.ProgramBlock(self.build_current_scope())
         return ast.ForLoopStatement(indexset_ast, loop_parameter_ast, body_ast)
 
+    def build_annotation(self, annotation: Annotation) -> ast.Annotation:
+        """Use the custom serializers to construct an annotation object."""
+        for namespace in iter_namespaces(annotation.namespace):
+            if (serializer := self.annotation_serializers.get(namespace, None)) is not None and (
+                payload := serializer.dump(annotation)
```

Comment:
Are there any checks we should do on the payload to make sure it's supported by QASM? Disclaimer: I haven't looked at the QASM spec to see what's supported.